src/include/transaction/transaction_manager.h

```diff
@@ -75,6 +75,21 @@ class TransactionManager {
    */
   TransactionQueue CompletedTransactionsForGC();
 
+  /**
+   * Adds the action to a buffered list of deferred actions.  This action will
+   * be triggered no sooner than when the epoch (timestamp of oldest running
+   * transaction) is more recent than the time this function was called.
+   * @param a functional implementation of the action that is deferred
+   */
+  void DeferAction(Action a);
```

Comment:
Should this be a collection of actions? Transactions can have lots of actions and they can be safely triggered at once using one timestamp. Calling this function in a loop is unnecessary.

Reply:
I think this is the best interface for epoch deferred actions (abort & commit actions don't go through this call).  While I think transactions can have lots of actions, I think the majority will either occur in bulk at abort or commit or be added piecemeal during a transaction's lifetime and be hard to aggregate into a group deferral without more coupling.

Is there a specific use case you have in mind?

Reply:
Sure. We can add group of actions later if needed.